extensions/wikia/Chat2/js/emoticons.js

```diff
@@ -35,49 +35,47 @@ WikiaEmoticons.doReplacements = function (text, emoticonMapping) {
 	$().log('Processing any emoticons... ');
 
 	var imgUrlsByRegexString = emoticonMapping.getImgUrlsByRegexString(),
-		regexString,
-		imgSrc,
-		numIters,
 		origText,
 		regex,
-		glyphUsed,
-		buildTagFunc;
-
-	for (regexString in imgUrlsByRegexString){
-		// Empty string for emote icons crash Chat so ignore them
-		if (regexString === '') {
-			continue;
-		}
+		buildTagFunc,
+		maxEmoticons = 5,
+		combinedRegex = Object.keys(imgUrlsByRegexString)
+			.map(function (key) {
+					return key.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
```

Comment:
You can move `/[-[\]{}()*+?.,\\^$|#\s]/g` regexp construction outside of map and create it only once and reuse it.